### PR TITLE
[Feature] Modular

### PR DESCRIPTION
**Pull request by Orrison**
Sets up the structure for us to do Module development. Currently staying on the Filament alpha and Livewire v2 as I was having trouble with the Beta. Will be reaching out to the core team tomorrow to address that issue.

But this should be good to merge **after** the first PR merges.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,4 @@ public/js/
 /.phpstorm.meta.php
 /phpstan.neon
 /.phpunit.result.cache
+app-modules/**/vendor
```
Comment (dgoetzit) [inline .gitignore:27]: For the purpose of the modules being individual units, we should have this within each module and add it to the scaffolding generated with the creation of each new module.

File: app-modules/case-module/database/migrations/2023_03_06_000044_add_relationship_fields_to_case_update_items_table.php
Comment (dgoetzit) [inline app-modules/case-module/database/migrations/2023_03_06_000044_add_relationship_fields_to_case_update_items_table.php:1]: Just a note that we can remove these extra migrations and put this into the original migration files.

File: app/Http/Controllers/Admin/CaseItemController.php
Comment (dgoetzit) [inline app/Http/Controllers/Admin/CaseItemController.php:1]: Is this being moved in a separate PR?

File: app/Http/Controllers/Admin/CaseItemPriorityController.php
Comment (dgoetzit) [inline app/Http/Controllers/Admin/CaseItemPriorityController.php:1]: Is this being moved in a separate PR?

File: app/Http/Controllers/Admin/CaseItemStatusController.php
Comment (dgoetzit) [inline app/Http/Controllers/Admin/CaseItemStatusController.php:1]: Is this being moved in a separate PR?

File: app/Http/Controllers/Admin/CaseItemTypeController.php
Comment (dgoetzit) [inline app/Http/Controllers/Admin/CaseItemTypeController.php:1]: Is this being moved in a separate PR?

File: app/Http/Controllers/Admin/CaseUpdateItemController.php
Comment (dgoetzit) [inline app/Http/Controllers/Admin/CaseUpdateItemController.php:1]: Is this being moved in a separate PR?

**Review by dgoetzit — APPROVED** [review]
This looks good - I have some questions but nothing blocking from moving forward.

File: app/Http/Controllers/Admin/CaseItemController.php
Comment (Orrison) [reply app/Http/Controllers/Admin/CaseItemController.php:1]: Yeah it will be removed

File: app/Http/Controllers/Admin/CaseItemPriorityController.php
Comment (Orrison) [reply app/Http/Controllers/Admin/CaseItemPriorityController.php:1]: Yeah it will be removed

File: app/Http/Controllers/Admin/CaseItemStatusController.php
Comment (Orrison) [reply app/Http/Controllers/Admin/CaseItemStatusController.php:1]: Yeah it will be removed

File: app/Http/Controllers/Admin/CaseItemTypeController.php
Comment (Orrison) [reply app/Http/Controllers/Admin/CaseItemTypeController.php:1]: Yeah it will be removed

File: app/Http/Controllers/Admin/CaseUpdateItemController.php
Comment (Orrison) [reply app/Http/Controllers/Admin/CaseUpdateItemController.php:1]: Yeah it will be removed
